winter/schema/type_hint.py

```diff
@@ -0,0 +1,188 @@
+
+TYPE_DECIMAL = openapi.TYPE_STRING if rest_settings.COERCE_DECIMAL_TO_STRING else openapi.TYPE_NUMBER
+TYPE_NONE = 'none'
+_resolvers_by_type: typing.Dict[typing.Type, typing.Tuple[typing.Callable, typing.Optional[typing.Callable]]] = {}
+
+
+def register(*types: typing.Tuple[typing.Type], checker=None):
+    def wrapper(func):
+        for type_ in types:
+            _resolvers_by_type[type_] = func, checker
+        return func
+
+    return wrapper
+
+
+@dataclasses.dataclass
+class YASGTypeInfo:
+    type_: str
+    format_: typing.Optional[str] = None
+    child: typing.Optional['YASGTypeInfo'] = None
+    nullable: bool = False
+    properties: typing.Dict[str, 'YASGTypeInfo'] = dataclasses.field(default_factory=OrderedDict)
+    enum: list = None
+
+    def as_dict(self):
+        data = {
+            'type': self.type_,
+        }
+        if self.format_ is not None:
+            data['format'] = self.format_
+
+        if self.child is not None:
+            data['items'] = self.child.as_dict()
+
+        if self.nullable:
+            data['x-nullable'] = True
+
+        if self.enum is not None:
+            data['enum'] = self.enum
+
+        if self.properties:
+            data['properties'] = {key: value.as_dict() for key, value in self.properties.items()}
+
+        return data
+
+
+# noinspection PyUnusedLocal
+@register(bool)
+def bool_resolver(hint_class) -> YASGTypeInfo:
+    return YASGTypeInfo(openapi.TYPE_BOOLEAN)
+
+
+# noinspection PyUnusedLocal
+@register(NoneType)
+def none_resolver(hint_class) -> YASGTypeInfo:
+    return YASGTypeInfo(TYPE_NONE)
+
+
+# noinspection PyUnusedLocal
+@register(int)
+def int_resolver(hint_class) -> YASGTypeInfo:
+    return YASGTypeInfo(openapi.TYPE_INTEGER)
+
+
+# noinspection PyUnusedLocal
+@register(str)
+def str_resolver(hint_class) -> YASGTypeInfo:
+    return YASGTypeInfo(openapi.TYPE_STRING)
+
+
+# noinspection PyUnusedLocal
+@register(float)
+def float_resolver(hint_class) -> YASGTypeInfo:
+    return YASGTypeInfo(openapi.TYPE_NUMBER)
+
+
+# noinspection PyUnusedLocal
+@register(dict)
+def dict_resolver(hint_class) -> YASGTypeInfo:
+    return YASGTypeInfo(openapi.TYPE_OBJECT)
+
+
+# noinspection PyUnusedLocal
+@register(decimal.Decimal)
+def decimal_resolver(hint_class) -> YASGTypeInfo:
+    return YASGTypeInfo(TYPE_DECIMAL, openapi.FORMAT_DECIMAL)
+
+
+# noinspection PyUnusedLocal
+@register(uuid.UUID)
+def uuid_resolver(hint_class) -> YASGTypeInfo:
+    return YASGTypeInfo(openapi.TYPE_STRING, openapi.FORMAT_UUID)
+
+
+# noinspection PyUnusedLocal
+@register(datetime.datetime)
+def datetime_resolver(hint_class) -> YASGTypeInfo:
+    return YASGTypeInfo(openapi.TYPE_STRING, openapi.FORMAT_DATETIME)
+
+
+# noinspection PyUnusedLocal
+@register(datetime.date)
+def date_resolver(hint_class) -> YASGTypeInfo:
+    return YASGTypeInfo(openapi.TYPE_STRING, openapi.FORMAT_DATE)
+
+
+# noinspection PyUnusedLocal
+@register(list, tuple, collections.Iterable)
```

Comment:
Iterable matches such types as `str`, `bytes`, is it desired behavior?

Reply:
str not, because it was registered above
I forgot register bytes (I'll do it)
So registering the str and bytes we solve the problem how to separete str and containers like list and tuple
If we set Iterable[str] - we always want to have have array of strings not one string